pkg/services/thumbs/service.go

```diff
@@ -284,3 +319,59 @@ func (hs *thumbService) getDashboardId(c *models.ReqContext, uid string) (int64,
 
 	return query.Result.Id, nil
 }
+
+func (hs *thumbService) runOnDemandCrawl(parentCtx context.Context, theme models.Theme, mode CrawlerMode, kind models.ThumbnailKind, authOpts rendering.AuthOpts) {
+	crawlerCtx, cancel := context.WithTimeout(parentCtx, hs.scheduleOptions.maxCrawlDuration)
+	defer cancel()
+
+	// wait for at least a minute after the last completed run
+	interval := time.Minute
+	err := hs.lockService.LockAndExecute(crawlerCtx, hs.crawlLockServiceActionName, interval, func(ctx context.Context) {
+		if err := hs.renderer.Run(crawlerCtx, authOpts, mode, theme, kind); err != nil {
+			hs.log.Error("On demand crawl error", "mode", mode, "theme", theme, "kind", kind, "userId", authOpts.UserID, "orgId", authOpts.OrgID, "orgRole", authOpts.OrgRole)
+		}
+	})
+
+	if err != nil {
+		hs.log.Error("On demand crawl lock error", "err", err)
+	}
+}
+
+func (hs *thumbService) runScheduledCrawl(parentCtx context.Context) {
+	crawlerCtx, cancel := context.WithTimeout(parentCtx, hs.scheduleOptions.maxCrawlDuration)
+	defer cancel()
+
+	authOpts := rendering.AuthOpts{
+		OrgID:   0,
+		UserID:  0,
+		OrgRole: models.ROLE_ADMIN,
```

Comment:
TODO: lets try with `VIEWER`

Reply:
tested - dashboards are unaccessible to _editor_ and _viewer_ org roles with `id & orgId === 0`